backend/server.js

```diff
@@ -961,10 +955,37 @@ app.get("/qInstances/:qiid/:filename", function(req, res) {
                 if (!_(info).has("clientFiles")) {
                     return sendError(res, 500, "Question does not have clientFiles, qid: " + fileInfo.qid);
                 }
-                if (!_(info.clientFiles).contains(fileInfo.filename)) {
-                    return sendError(res, 404, "Access denied to '" + fileInfo.filename + "' for qid: " + fileInfo.qid);
+
+                if (_(info).has("clientTemplates") && _(info.clientTemplates).contains(fileInfo.filename)) {
+                    // File is template which needs to be rendered
+                    fs.readFile(path.join(fileInfo.root, fileInfo.filePath), function(err, data) {
```

Comment:
I noticed the same, but this should be safe because we first check whether the filename is in `clientTemplates`. The only risk here is that we send different status codes for the file not existing and the file not being in `clientFiles`/`clientTemplates`, which would allow an adversary to determine the existence of files on the filesystem.